### PR TITLE
KS-471: update capabilities of nodes in capability registry

### PR DESCRIPTION
**Pull request by krehermann**
[KS-471](https://smartcontract-it.atlassian.net/browse/KS-471)

This change adds support in the  deployment package to both update and append capabilities to existing nodes.

The capability registry doesn't directly support appending capabilities. It only supports setting them.

Therefore to achieve the end, we have to read the existing state merge the desired capabilities with the existing one.

There are distinct operations to fetching, merging, and setting. I modeled them independently so that the can be composed in the `AppendNodeCapabilities` func.

In order to test this, I introduced a `test` package that sets up the registry contract in a given state. A side effect of this new package is that some of the struct and funcs needed to become public, and the PR is littered with capitalization changes.

Most of this change is the testing, bookkeeping between the existing contract state and the desired state, and the private->public renaming.

[KS-471]: https://smartcontract-it.atlassian.net/browse/KS-471?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: integration-tests/deployment/keystone/update_nodes.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+				Signer:              node.Signer,
+			})
+		}
+	}
+
+	return out, nil
+}
+
+func CapabilityID(c kcr.CapabilitiesRegistryCapability) string {
+	return fmt.Sprintf("%s_%s_%d", c.LabelledName, c.Version, c.CapabilityType)
```
Comment (patrickhuie19) [inline integration-tests/deployment/keystone/update_nodes.go]: Is the intent for this to be a different type of CapabilityID from [this](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/keystone/CapabilitiesRegistry.sol#L170) ?

File: integration-tests/deployment/keystone/update_nodes_test.go [hunk 2]
```diff
@@ -0,0 +1,516 @@
...
+						},
+					},
+					Chain:    chain,
+					Registry: nil, // set in test to ensure no conflicts
+				},
+			},
+			want: &kslib.UpdateNodesResponse{
+				NodeParams: []kcr.CapabilitiesRegistryNodeParams{
+					{
+						NodeOperatorId:      1,
```
Comment (patrickhuie19) [inline integration-tests/deployment/keystone/update_nodes_test.go:214]: /nit `ID` vs `Id` for our linting rules

**Review by patrickhuie19 — DISMISSED** [review]

File: integration-tests/deployment/keystone/update_nodes_test.go (hunk 2 above)
Comment (krehermann) [reply integration-tests/deployment/keystone/update_nodes_test.go:214]: generated struct field names from solidity

File: integration-tests/deployment/keystone/update_nodes.go (hunk 1 above)
Comment (krehermann) [reply integration-tests/deployment/keystone/update_nodes.go]: thanks for pointing out. fixed and new comment

File: integration-tests/deployment/keystone/changeset/append_node_capbilities.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
+package changeset
```
Comment (bolekk) [inline integration-tests/deployment/keystone/changeset/append_node_capbilities.go:1]: nit: typo in file name

File: integration-tests/deployment/keystone/changeset/update_node_capabilities.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	if len(req.NopToNodes) == 0 {
+		return fmt.Errorf("nopToNodes is empty")
+	}
+	if req.Registry == nil {
+		return fmt.Errorf("registry is nil")
+	}
+	return nil
+}
+
+// UpdateNodeCapabilibity sets the capabilities of the node to the new capabilities.
```
Comment (bolekk) [inline integration-tests/deployment/keystone/changeset/update_node_capabilities.go:34]: nit: typo

**Review by bolekk — APPROVED** [review]

**Review by vyzaldysanchez — APPROVED** [review]

**Pull request merged**